pytorch_lightning/trainer/training_loop.py

```diff
@@ -412,8 +412,8 @@ def run_training_epoch(self):
         train_dataloader = self.train_dataloader
 
         # on TPU we have to wrap it under the ParallelLoader
-        if self.use_tpu and self.tpu_id is None:
-            device = xm.xla_device()
+        if self.use_tpu:
+            device = xm.xla_device(self.tpu_id)
```

Comment:
what if `self.tpu_id` is None?

Reply:
Sorry. Missed that out. Have added a condition. :]

Reply:
where is the condition? i still don't see where you check for None

Reply:
as mentioned in concurrent PR, `None` is allowed option